galley/pkg/config/analysis/local/analyze.go

```diff
@@ -21,36 +21,49 @@ import (
 
 	"istio.io/istio/galley/pkg/config/analysis"
 	"istio.io/istio/galley/pkg/config/analysis/diag"
+	"istio.io/istio/galley/pkg/config/collection"
 	"istio.io/istio/galley/pkg/config/event"
 	"istio.io/istio/galley/pkg/config/meshcfg"
 	"istio.io/istio/galley/pkg/config/processing/snapshotter"
 	"istio.io/istio/galley/pkg/config/processing/transformer"
 	"istio.io/istio/galley/pkg/config/processor"
 	"istio.io/istio/galley/pkg/config/processor/transforms"
 	"istio.io/istio/galley/pkg/config/schema"
+	"istio.io/istio/galley/pkg/config/scope"
 	"istio.io/istio/galley/pkg/config/source/kube/apiserver"
 	"istio.io/istio/galley/pkg/config/source/kube/inmemory"
 	"istio.io/istio/galley/pkg/source/kube/client"
 )
 
 const domainSuffix = "svc.local"
 
+// Patch table
+var (
+	apiserverNew = apiserver.New
+)
+
 // SourceAnalyzer handles local analysis of k8s and file based event sources
 type SourceAnalyzer struct {
 	m                    *schema.Metadata
 	sources              []event.Source
 	analyzer             analysis.Analyzer
 	transformerProviders transformer.Providers
+
+	// Which collections are used by this analysis
+	// Derived from the specified analyzer and transformer providers
+	inputCollections map[collection.Name]bool
```

Comment:
This one still have bool instead of struct{} :)

Reply:
Fixed. Sigh.

I did a quick grep through the rest of Istio and incredibly I don't see *either* style being used *anywhere*. Is Istio really just written without set semantics, or is there another lib or something in use?

Reply:
it map[?]struct{} should be pretty prevalent. The source base is a mixture of both but the struct{} had perf benefits for CPU-intensive operations, when we were looking at this. It is also less confusing, as bool can be misconstrued as semantically important.